### DISCUSSION TITLE
init script error

### DISCUSSION THREAD

**Discussion by arki82** [Gulivertx/cbatmo#69] [General]
Hi thanks for this project. Somehow the init script seems to fail on my raspberry. The manual start works, but there is a problem with the working directory: 
"Process: 5165 ExecStart=/usr/bin/yarn start (code=exited, status=200/CHDIR)"

```
pi@raspberrypi:~/cbatmo $ sudo cp rpi/cbatmo.service /etc/systemd/system/
pi@raspberrypi:~/cbatmo $ sudo systemctl enable cbatmo.service
Created symlink /etc/systemd/system/multi-user.target.wants/cbatmo.service \u2192 /etc/systemd/system/cbatmo.service.
pi@raspberrypi:~/cbatmo $ sudo systemctl restart cbatmo.service
pi@raspberrypi:~/cbatmo $ systemctl status cbatmo.service
\u25cf cbatmo.service - CBatmo WEB server
   Loaded: loaded (/etc/systemd/system/cbatmo.service; enabled; vendor preset: e
   Active: activating (auto-restart) (Result: exit-code) since Sun 2021-03-28 13
  Process: 5165 ExecStart=/usr/bin/yarn start (code=exited, status=200/CHDIR)
 Main PID: 5165 (code=exited, status=200/CHDIR)

pi@raspberrypi:~/cbatmo $ yarn start
yarn run v1.22.10
$ node server.js
Server running on http://localhost:3000 as production
```

**Comment by alpew** [comment]
@arki82 I get exactly the same error. DId you manage to sort it?

**Comment by Gulivertx** [comment]
Hi,

Could you give us what there is inside this file 
/etc/systemd/system/cbatmo.service

**Reply by alpew** [reply]
Hi.

I reformatted the Pi SD card and put a fresh Rspbian image on this morning. I tried the ./deploy.sh from my Mac and also tried copying a zip file across by used stick to the Pi and locally installing and still getting:

pi@raspberrypi:~/Downloads/cbatmo $ sudo systemctl status cbatmo.service
* cbatmo.service - CBatmo WEB server
   Loaded: loaded (/etc/systemd/system/cbatmo.service; enabled; vendor preset: e
   Active: activating (auto-restart) (Result: exit-code) since Mon 2021-07-05 14
  Process: 1490 ExecStart=/usr/bin/yarn start (code=exited, status=200/CHDIR)
 Main PID: 1490 (code=exited, status=200/CHDIR)


I've attached the files as I went through the instructions to deploy from the Mac that also includes the contents of cbatmo.service

Apologies in advance if its a simple user error. I am not a programmer and don't really understand what I am doing to be honest. I've gone through the instructions step by step to install. 

Cheers

Al
[Pi error.txt](https://github.com/Gulivertx/cbatmo/files/6764677/Pi.error.txt)
[cbatmo.service.txt](https://github.com/Gulivertx/cbatmo/files/6764678/cbatmo.service.txt)
[git install.txt](https://github.com/Gulivertx/cbatmo/files/6764679/git.install.txt)

**Comment by alpew** [comment]
Hi 

An update over previous post,

I've checked nodejs and yarn on my mac and Pi to make sure all are updated to latest version.

I deployed from the mac to the Pi over my LAN using
> ./deploy.sh
The attached file 'install-from-mac-terminal.txt" includes the terminal log from running the deploy command.

The wifi error is due to the fact both the MAC and Pi are connected to the LAN via an Ethernet cable to a LAN switch instead of WiFi.

I've also attached the latest cbatmo.service file from the Pi and also the pi-systemctl.txt file which details the response from the systemctl status command.

I hope this is useful.

Al

[rpi-cbatmo.service.txt](https://github.com/Gulivertx/cbatmo/files/6765547/rpi-cbatmo.service.txt)
[rpi-systemctl.txt](https://github.com/Gulivertx/cbatmo/files/6765548/rpi-systemctl.txt)
[install-from-mac-terminal.txt](https://github.com/Gulivertx/cbatmo/files/6765549/install-from-mac-terminal.txt)
